pkg/release/push.go

```diff
@@ -485,3 +487,39 @@ func (p *PushBuild) CopyStagedFromGCS(stagedBucket, version, buildVersion string
 
 	return nil
 }
+
+// StageLocalSourceTree creates a src.tar.gz from the Kubernetes sources and
+// uploads it to GCS.
+func (p *PushBuild) StageLocalSourceTree(buildVersion string) error {
+	workDir := os.Getenv("GOPATH")
+	if workDir == "" {
+		return errors.New("GOPATH is not set")
+	}
+
+	const src = "src.tar.gz"
+	tarball := filepath.Join(workDir, src)
+	logrus.Infof("Creating source tree tarball in %s", workDir)
+
+	if err := command.New(
+		"tar", "cfz", tarball,
+		fmt.Sprintf("--exclude=src/k8s.io/kubernetes/%s-*", BuildDir),
+		"-C", workDir,
+		"src",
+	).RunSilentSuccess(); err != nil {
+		return errors.Wrap(err, "create source tarball")
+	}
```

Comment:
Any reason not to use `archive/tar` here?

Reply:
I wasn't strongly opinionated but it just seemed that it will increase the verbosity of the code a lot. Do you prefer using the `archive/tar` package there?

Reply:
I generally prefer to use native functionality rather than exec'ing out to a command. However, these kind of optimizations can be introduced further down the line. Exec'ing out is certainly an improvement over the bash :)